src/components/forms/textInput.stories.mdx

```diff
@@ -0,0 +1,177 @@
+complete with accessibile labels, Icons, side labels, errorText, throttled events, and more.
+
+<Story name="Knobs" parameters={{ docs: { disable: true } }}>
+  <TextInput
+    label={text("label", "Label")}
+    errorText={text("errorText", undefined)}
+    required={boolean("required", undefined)}
+    type={text("type", undefined)}
+    onChange={action("changed")}
+    IconRight={boolean("Show IconRight", undefined) ? CaretDownIcon : undefined}
+    IconLeft={boolean("Show IconLeft", undefined) ? CaretDownIcon : undefined}
+    before={text("before", undefined)}
+    after={text("after", undefined)}
+  />
+</Story>
+
+<Preview isExpanded>
+  <>
+    <TextInput />
+    <br />
+    <TextInput
+      label="First Name"
+      required
+      errorText="Custom error text"
+      placeholder="Placeholder text"
+      IconRight={CaretDownIcon}
+      before="$"
+      after=".com"
+    />
+  </>
+</Preview>
+
+<Info type="info">
+  Additional props will be spread on the `input` element.
+</Info> 
+
+<Props of={TextInput} />
+
+## Example Layout
+
+<Preview>
+  <Story name="Example Layout">
+    <>
+      <Flex>
+        <FlexItem>
+          <TextInput label="First Name" required />
+        </FlexItem>
+        <FlexItem>
+          <TextInput label="Last Name" required />
+        </FlexItem >
+      </Flex>
+      <Flex>
+        <FlexItem>
+          <TextInput
+            type="email"
+            label="Email"
+            errorText="Enter a valid email address"
+            placeholder="e.g. anand@cbinsights.com"
+          />
+        </FlexItem>
+        <FlexItem>
+          <TextInput type="password" label="Password" />
+        </FlexItem>
+        <FlexItem>
+          <TextInput type="number" label="Zipcode" after="90210" />
+        </FlexItem>
+      </Flex>
+    </>
+  </Story>
+</Preview>
+
+### Throttle
+
+The prop `onThrottleChange` fires a callback that throttles less than 5 characters,
+and debounces 5 or more characters.
+https://www.peterbe.com/plog/how-to-throttle-and-debounce-an-autocomplete-input-in-react
+
+View this story in Canvas, in the action tab, to see how the callback is throttled.
+
+<Preview>
+  <Story name="Throttle">
+    <TextInput
+      throttleOnChange={action("changed")}
+    />
+  </Story>
+</Preview>
+
+
+### Controlled Input
+
+<Story name="Controlled Input">
+  <TextInput
+    value="Controlled"
+  />
+</Story>
+
+### Icons
+
+<Story name="Icons">
+  <TextInput
+    IconLeft={CaretDownIcon}
+    IconRight={CaretDownIcon}
+  />
+</Story>
+
+### Side label
+
+<Story name="Before & After">
+  <TextInput
+    before="before"
+    after="after"
+  />
+</Story>
+
+### Side Labels + Icon
+
+<Story name="Side + Icon">
+  <TextInput
+    label="Label"
+    placeholder="placeholder"
+    before="before"
+    after="after"
+    IconLeft={CaretDownIcon}
+    IconRight={CaretDownIcon}
+  />
+</Story>
+
+### Misc
+
+Can be used with InputGroup (why don't these stretch full width?)
```

Comment:
was this an unaddressed TODO?

Reply:
This is possibly more of a question for you I think? IconInput, DecoratedInput, etc, all stretch to be 100% width, except InputGroup. Sort of wondering if it's expected that this is different, or if it should be the same.

I'm glad I wrote this because I completely forgot about it.

Reply:
the outer border for an input group is the root element of the `InputGroup` component. This element is set to `min-content` because the last `input` or `select` element in the group does not grow to fill remaining space.